src/librustc/ty/layout.rs

```diff
@@ -2023,7 +2023,14 @@ where
 {
     fn for_variant(this: TyLayout<'tcx>, cx: &C, variant_index: VariantIdx) -> TyLayout<'tcx> {
         let details = match this.variants {
-            Variants::Single { index } if index == variant_index => this.details,
+            Variants::Single { index }
+                // If all variants but one are uninhabited, the variant layout is the enum layout.
+                if index == variant_index &&
+                // Don't confuse variants of uninhabited enums with the enum itself.
```

Comment:
Maybe reference https://github.com/rust-lang/rust/issues/69763 to explain why that's a problem